[Refactor] Use `Phaser.Scale.Orientation` for `isLandscapeMode`

flx-sta opened the pull request:
## What are the changes the user will see?

n/a

## Why am I making these changes?

Closes #1068 

## What are the changes from a developer perspective?

- `isLandscapeMode()` now takes an argument that must include an instance of the `Phaser.Scale.ScaleManager`
- Updated `create-test-boilerplate` to also take `Utils` as argument and be easier to extend in the future
- Split ~~`utils.test.ts`~~ into `random-utils.test.ts` and `string-utils.test.ts`
- Add `app-utils.test.ts`

## Screenshots/Videos

### Portrait | IPhone 16, iOS 18.0

<details><summary>Details</summary>
<p>

| Dark | Light |
|--------|--------|
| <img width="418" alt="image" src="https://github.com/user-attachments/assets/b94f592f-6519-4780-a843-4654a9247b1f" /> | <img width="420" alt="image" src="https://github.com/user-attachments/assets/f703219e-a75f-4eb4-8cca-ed0ecd21084c" /> | 

</p>
</details> 

### Landscape | IPhone 16, iOS 18.0

<details><summary>Details</summary>
<p>

| Dark | Light |
|--------|--------|
| <img width="861" alt="image" src="https://github.com/user-attachments/assets/6b31f1c2-cd7b-4915-9e86-1ffaf2089c18" /> | <img width="843" alt="image" src="https://github.com/user-attachments/assets/9c49780a-611d-459d-b5c8-ebb651c36518" /> | 

</p>
</details> 

## How to test the changes?

Run the tests

Emulate a touchscreen device in your browser and see if the correct orientation name is displayed in the general settings for the `Move Touch Controls` option

## Checklist

- [x] Otherwise: **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs?
    - **Maybe not as I've also updated the `create-test-boilerplate` script**
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes manually?
- [x] Are all unit tests still passing? (`npm run test:silent`)
  - [x] Have I created new automated tests (`npm run test:create`) or updated existing tests related to the PR's changes?
- [x] Have I provided screenshots/videos of the changes (if applicable)?
  - [x] Have I made sure that any UI change works for both UI themes (dark and light)?

flx-sta committed:
split `utils.test` into `random-utils` and `string-utils` test

flx-sta committed:
make isLandscapeMode use phaser orientation

flx-sta committed:
add test-coverage for app-utils.isLandscapeMode

flx-sta committed:
optimize app-utils

flx-sta committed:
revert generalSettingsUiItems back to array

from function

Tempo-anon committed:
[P2] Transistor is 1.3x instead of 1.5x (#1096)

Transistor is 1.3x instead of 1.5x

Tempo-anon committed:
[AI][Balance] Enemy AI trainers will now tera (#1084)

* Enemy AI trainers will now tera

* Delete commented out code

Tempo-anon reviewed (dismissed):
Code looks good to me, though the refactoring stuff feels like it could be its own PR

Tempo-anon committed:
[Refactor] Rename SeedTag to SeededTag (#1095)

* Rename SeedTag to SeededTag

* Forgot to add get-battler-tag.ts

flx-sta committed:
[Checkup][Ability] Add Steadfast Ability Checkup (#1069)

* Reduce `loaded locale` noise in tests

- it can still be printed by setting "VITE_I18N_DEBUG=1"

* Add 2 matchers

- to have flinched matcher
- to have stat stage matcher

* Add tsdoc to FlinchStatStageChangeAbAttr

* fix [Bug] Flinching Can Occur After a Pokemon Has Moved #529

- Make FlinchStatStageChangeAbAttr only apply if the pokemon hasn't acted yet

* Add Steadfast tests

* Fix: FlinchStatStageChangeAbAttr application lifecycle

the tag was originally applied immediately after flinching, which cause the attribute to also be triggered when the flinching occured AFTER the owner already acted. Now it's only applied in lapse and only if the lapse type ise "pre-move".

* update Steadfast tests

* Update test/abilities/steadfast.test.ts

Co-authored-by: Tempoanon <163687446+Tempo-anon@users.noreply.github.com>

* Update test/abilities/steadfast.test.ts

Co-authored-by: Tempoanon <163687446+Tempo-anon@users.noreply.github.com>

* update Steadfast tests

- add test for "Inner Focus"
- add tests for "Inner Focus" being overriden by enemy ability

* update Steadfast tests

fix order of some expect checks to match tag check better

* add 2 Steadfast tests (1 todo)

- add coverage for "Shield Dust" ability
- add todo for Covert Cloak Item (not implemented yet)

* Update test/matchers/to-have-flinched.ts

Co-authored-by: PigeonBar <56974298+PigeonBar@users.noreply.github.com>

* Update test/abilities/steadfast.test.ts

Co-authored-by: PigeonBar <56974298+PigeonBar@users.noreply.github.com>

* Update src/data/battler-tags/flinched-tag.ts

Co-authored-by: PigeonBar <56974298+PigeonBar@users.noreply.github.com>

* Update test/matchers/to-have-stat-stage.ts

* add flinch test

* chore: fix matchers.setup imports

* Update src/data/abilities/ab-attrs/flinch-stat-stage-change-ab-attr.ts

Co-authored-by: innerthunder <168692175+innerthunder@users.noreply.github.com>

* rename `toHaveFlinched` to `toHaveBattlerTagType`

* add `toHaveAbilityApplied` matcher

* update move-phase console.log

now includes the pokemon name with affix

* Update matchers output

* Add undefined check to toHaveUsedMove matcher

---------

Co-authored-by: Tempoanon <163687446+Tempo-anon@users.noreply.github.com>
Co-authored-by: PigeonBar <56974298+PigeonBar@users.noreply.github.com>
Co-authored-by: innerthunder <168692175+innerthunder@users.noreply.github.com>

flx-sta commented:
> Code looks good to me, though the refactoring stuff feels like it could be its own PR

Yeah… it was one of those too far in to go back now situations

Tempo-anon committed:
[Refactor] Remove deprecated functions in phaseInterceptor (#1101)

* Remove deprecated functions in phaseInterceptor

* Remove phaseInterceptor.phaseFrom

innerthunder committed:
[Refactor] Remove deprecated `nextCommandPhaseQueue` and deferred phase scheduling (#1103)

* Remove `nextCommandPhaseQueue` and deferred phase scheduling

* Fix misleading `eager` description

dependabot[bot] committed:
Bump vite from 6.2.6 to 6.3.4 (#1106)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.2.6 to 6.3.4.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.3.4/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.3.4
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

PigeonBar committed:
[Test] Fix a flaky Download test (#1102)

flx-sta committed:
Move PRIMAL_WEATHER_TYPES into weather const

DayKev committed:
[Refactor] Consolidate arena tag functions (#1074)

* Remove `getTagOnSide` and `findTagOnSide`

* Rename `getTag` to `findTag` and `findTags` to `getTags`

* Add TSDocs to `hasTag` and use `hasTag` over `findTag` when applicable

* `findTag` no longer accepts Classes

* Remove `instanceof` + leftover `TODO`

* Add explanation to todo

* Fix merge issue

* Apply changes to tests and fix import

* Update tests

* Fix Trick Room test check

* Modify tag search functions

* `Arena#getTags` now has a return type hint of `| undefined`

Remove `instanceof EntryHazardTag` in `trainer.ts`

* Remove `instanceof EntryHazardTag` in `arena.ts`

---------

Co-authored-by: flx-sta <50131232+flx-sta@users.noreply.github.com>

Tempo-anon committed:
[Beta] Nemona has proper tera (#1105)

* Nemona has proper tera

* merge beta and pr comments

flx-sta committed:
[Dev][Refactor] Reduce api noise (#1108)

* add VITE_API_DEBUG env

* Update api ping & logging

- make use of "VITE_API_DEBUG".
- Prevent default logging during ping
- Don't use `getGameTitleStats()` for pinging anymore

* update api

- add handling for server not connected
- add more optional logging

* update api.test

flx-sta committed:
[Dev][Refactor] Add `-constants` suffix to constants files (#1112)

add `-constants` suffix to `src/constants/` files

Tempo-anon committed:
[Misc] Add exp formula for gens 1-7 (#1075)

* Add exp formula for gens 1-7

* Add export

* Add `Battle#isTrainerBattle`, reformat functions, update comments (#1088)

* Document some more exp stuff

* Review comments

---------

Co-authored-by: NightKev <34855794+DayKev@users.noreply.github.com>
Co-authored-by: flx-sta <50131232+flx-sta@users.noreply.github.com>

DayKev committed:
[Misc] Fix ME `build()` functions (#1118)

Fix ME `build()` functions

DayKev committed:
[Refactor] Add and implement `coerceArray` utility function (#1090)

* [Refactor] Add and implement `coerceArray` utility function

* Use less precise method to resolve type casting

* Missed some places to use `coerceArray`

flx-sta commented:
Well, that rebase didn't go well
